lib/pengine/rules_alerts.c

```diff
@@ -27,6 +27,13 @@ get_meta_attrs_from_cib(xmlNode *basenode, pcmk__alert_t *entry,
                       NULL, FALSE, now, NULL);
     crm_time_free(now);
 
+    value = g_hash_table_lookup(config_hash, PCMK_XA_ALERT_ENABLED);
+    if ((value != NULL) && !crm_is_true(value)) {
+        // No need to continue unpacking
+        entry->enabled = false;
```

Comment:
This is my preferred approach.

We *could* avoid adding the `pcmk__alert_t:enabled` member and instead set the `entry->flags` to `pcmk__alert_none` if `enabled=false`. However, we would do that in `unpack_alert_filter()`, which requires that we finish unpacking the alert that we're not going to use.

Another option would be to set `entry->flags = pcmk__alert_none` here, and check that before calling `unpack_alert_filter()`. That would work since the default value of `entry->flags` is not `pcmk__alert_none` (it's nodes|fencing|resources). But that feels less clear to me since `unpack_alert_filter()` is what's really supposed to set those flags.

Reply:
With your implementation, we don't really need to track disabled in the object at all, we just need get_meta_attrs_from_cib() and unpack_alert() to return success/skip

Reply:
Oh you're right! When I started working on this and added that member, I thought it was going to be a little bit more complicated than it turned out to be.